TestProject_Calc/CalculatorPage.cs

```diff
@@ -1,7 +1,184 @@
+            {
+                property.SetValue(inputData, expectedReplacement);
+            }
+
+            return inputData;
+        }
+
+        public CalculatorValues GenerateExpectedErrorDTO(string depositValue = "1", string rateValue = "1", string term = "1", string day = "1", string month = "January", string year = "2014", FinancialYearInput? financialYear = FinancialYearInput.full)
+        {
+            var inputData = new CalculatorValues
+            {
+                DepositValue = depositValue,
+                RateValue = rateValue,
+                Term = term,
+                Day = day,
+                Month = month,
+                Year = year,
+                FinancialYear = financialYear,
+                InteresetEarned = "0.00",
+                Income = "0.00",
+                EndDate = DateTime.Now.ToString("dd/MM/yyyy")
+            };
+
+            return inputData;
+        }
+
+        public CalculatorValues EnterInputs_GetOutput(CalculatorValues inputData, string errorField = "", string expectedReplacement = "0")
+        {
+            EnterInputs(inputData);
+
+            if (String.IsNullOrEmpty(errorField)) ClickCalculate();
+            else ChangeExpectedDTO(inputData, errorField, expectedReplacement);
+
+            return GetActualDTO();
+        }
+
+        public void EnterInputs(CalculatorValues inputData)
+        {
+            InsertDepositAmmount(inputData.DepositValue);
+            InsertRateOfInterest(inputData.RateValue);
+            InsertInvestmentTerm(inputData.Term);
+            InsertStartDateDay(inputData.Day);
+            InsertStartDateMonth(inputData.Month);
+            InsertStartDateYear(inputData.Year);
+            ChoseFinancialYear(inputData.FinancialYear);
+        }
+
+        public CalculatorValues GetActualDTO()
+        {
+            static string value(IWebElement item) => item.GetAttribute("value");
+
+            var outputData = new CalculatorValues
+            {
+                DepositValue = value(DepositAmmountField),
+                RateValue = value(RateOfInterestField),
+                Term = value(InvestmentTermField),
+                Day = value(StartDateDay),
+                Month = value(StartDateMonth),
+                Year = value(StartDateYear),
+                FinancialYear = CheckFinancialYear(),
+                InteresetEarned = value(InteresetEarnedField),
+                Income = value(IncomeField),
+                EndDate = value(EndDateField)
+            };
+
+            return outputData;
+        }
+
+        public string? CompareValues(CalculatorValues inputData, CalculatorValues outputData)
+        {
+            var consoleLog = "";
+
+            foreach (PropertyInfo prop in calculatorProperty)
+            {
+                var input = prop.GetValue(inputData);
+                var output = prop.GetValue(outputData);
+
+                if (!input.Equals(output))
```

Comment:
Why: if `input` is null your code will throw exception, this one will return `false`
```suggestion
                if (input != output)
```